### ISSUE TITLE
use import_module instead of __import__

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#10008]
Since Python 3.3 it is [_requested_ to use importlib.import_module instead of `__import__`](https://docs.python.org/3/library/importlib.html#importlib.__import__).

The reason is not really given, but I guess there are reasons.

**Comment by Tushar-Gupta** [comment]
Does the fixing of this bug require all import statements in all Python files in this project to be changed to use importlib.import_module?

**Comment by Carreau** [comment]
Hi @Tushar-Gupta , 

> Does the fixing of this bug require all import statements in all Python files in this project to be changed to use importlib.import_module?

No far from that. If will likely affect only a couple of imports done dynamically using `__import__` and likely just a subset of the following lines:

```
$ grep -r __import__ **.py
IPython/core/completer.py:            isinstance(obj, getattr(__import__(module), class_name)))
IPython/core/completerlib.py:        m = __import__(mod)
IPython/core/debugger.py:            # needed by any code which calls __import__("__main__") after
IPython/core/extensions.py:                    __import__(module_str)
IPython/extensions/autoreload.py:        __import__(module_name)
IPython/external/qt_loaders.py:        mod = __import__(module_name)
IPython/lib/deepreload.py:original_import = builtin_mod.__import__
IPython/lib/deepreload.py:    saved_import = builtin_mod.__import__
IPython/lib/deepreload.py:    builtin_mod.__import__ = new_import
IPython/lib/deepreload.py:        builtin_mod.__import__ = saved_import
IPython/lib/deepreload.py:        # 'from . import' (or '__import__("")')
IPython/lib/deepreload.py:    """Replacement for __import__()"""
IPython/lib/deepreload.py:    # an empty module name: someone called __import__("") or
IPython/testing/decorators.py:        mod = __import__(module)
IPython/testing/plugin/ipdoctest.py:            module = __import__(modname)
IPython/utils/importstring.py:        module = __import__(package, fromlist=[obj])
IPython/utils/importstring.py:        return __import__(parts[0])
IPython/utils/rlineimpl.py:        _rl = __import__(_rlmod_name)
IPython/utils/shimmodule.py:        return __import__(self._mirror).__spec__
IPython/utils/shimmodule.py:        return dir(__import__(self._mirror))
IPython/utils/shimmodule.py:        mod = __import__(self._mirror)
docs/sphinxext/apigen.py:        self.root_module = __import__(package_name)
```

For each of these case  we would need to inspect what `__import__` does, and see if replacing it with `import_module` make sens. In some case I even expect `import_module` to simplify the code.

Also this is not really a bug. It is more code cleaning to use new features available only in python 3.3+ so there is no hurry.
